ユーザのフォローアンフォローができる

ku00 opened the pull request:
@orzup @alotofwe 

ユーザのフォロー/アンフォローをできるようにします。各ユーザのProfileHeaderにボタンを設置し、そこからフォロー/アンフォローが行えることを想定しています。
余力があればユーザリスト(All, Followers, Following)上でも行えるようにしたいです。
### 使うAPI
- https://github.com/alotofwe/sample_app-1/pull/36
- https://github.com/alotofwe/sample_app-1/pull/38
- https://github.com/alotofwe/sample_app-1/pull/46
### merge条件
- [x] @orzup or @alotofwe  が動作確認をすること
  - [x] フォロー/アンフォローボタンが表示されていること
    - ProfileHeader
    - All
    - Followers
    - Following
  - [x] フォローができること
  - [x] アンフォローができること
  - [x] フォロー状態が正しいこと
- [x] TravisCIがグリーンであること

ku00 committed:
[WIP]

ku00 committed:
follow, unfollowのapiをたたくメソッドを用意

メソッド名はapiのurlに合わせてる(けどどうなんだろ)

ku00 committed:
ボタンを設置

ku00 committed:
follow, unfollowで表示を切り替える

ku00 committed:
不要なidentifierを削除

ku00 committed:
follow, unfollowができる

ただし状態を保持していないためfollowからはじまるようになってる。
なので、確認したい場合は画面を切り替えずにボタンを2回タップする必要がある

ku00 commented:
API、フォロー状態を返してくれるパラメータを追加した

ku00 committed:
フォロー状態からボタンを切り替えて表示するようにした

ku00 committed:
ボタンのスタイルをメソッドにまとめた

ku00 committed:
ユーザリストにボタンを表示するまで

ku00 committed:
ボタンを押したときのアクションがうまく紐付いてなかったのを解消

ku00 committed:
ユーザリストからフォロー/アンフォローができる

ku00 commented:
**note:**
ログインしているユーザにもボタンが設置されてしまってるので後で非表示にするようにしたい

ku00 commented:
それとプロフィールとユーザリストでそれぞれ同じようなコードを書いてるから共通化して分離したい

alotofwe commented on Kakico/Classes/Controllers/ProfileHeaderViewController.swift:
```diff
@@ -17,6 +18,16 @@ class ProfileHeaderViewController: UIViewController {
         request(_selectUserId)
     }
 
+    @IBAction func followOrUnfollow(sender: UIButton) {
```
好みの問題だけれど，toggleFollowだと，切り替えている感じがあって私は好きです... :sweet_potato:

ku00 replied:
toggleいいですね、修正しておきます :cyclone:

alotofwe commented on Kakico/Classes/Models/User.swift:
```diff
@@ -1,7 +1,7 @@
 
 struct User {
-    var id: Int, name: String, icon: NSURL
+    var id: Int, name: String, icon: NSURL, following_flag: Bool
```
スネークケース :eyes:

ku00 replied:
あああ、忘れてたあああ。直します... :bow:

alotofwe commented:
PC版では自分自身はフォローできないようになっているけれど
iOS版でもそうしたほうがいいかな...?

ku00 commented:
> PC版では自分自身はフォローできないようになっているけれど
> iOS版でもそうしたほうがいいかな...?

そうしたい、けれどこのブランチだと自分を参照できないので別PRで対処してもいいかなあと考えてました。

alotofwe commented:
シミュレータで，merge条件通りの挙動をしていることを確認しました :ok_woman: 

> そうしたい、けれどこのブランチだと自分を参照できないので別PRで対処してもいいかなあと考えてました。

そういえばそうでした...
現在のmasterでは，もうkeychainにuser_idを入れているので
コンフリクト解消時に，このブランチで簡単に対応できるかなと思います

ので，コンフリクト解消後にお願いしてもいいですか...? :sweat_drops:

ku00 committed:
ボタン切り替えのアクション名を変更

ku00 commented:
はーい :ok_woman:

ku00 committed:
スネークケースを撲滅

ku00 commented:
レビューで指摘された箇所は修正したのでマージしますね :cat2:

ku00 commented:
じゃなくてdevelopmentをこっちに取り込んで、自分自身をフォローできないようにするタスクに取り掛かるんだ

ku00 committed:
Merge branch 'development' into follow-unfollow

ku00 committed:
コンフリクト解消

つらかった(´･_･`)
とりあえず削除したいUIObjectがあったときは、
それに紐付いてるID(これにはlayout周りのものも含まれる)をメモっておいて、
該当するIDのコード部分を消すって作業をやったらうまくいった気がするよ

ku00 committed:
自分自身のセルではボタンを非表示にする

とりあえずリストのみ

ku00 commented:
ボタン非表示、しなくていいユーザのボタンまで消してる...

ku00 committed:
ボタンの非表示を修正

自分自身のみ非表示にすることができた

ku00 committed:
自分のプロフィールではボタンを非表示にする

ついでにボタン初期化のリファクタリングもした

ku00 committed:
Merge branch 'development' into follow-unfollow

ku00 committed:
コンフリクト解消
